packages/box/box.ts

```diff
@@ -19,50 +21,75 @@ const defaultPath = "https://github.com:trufflesuite/truffle-init-default";
  *   - git@github.com:<org>/<repo>[#branch]
  *   - path to local folder (absolute, relative or ~/home)
  */
-const normalizeSourcePath = (url = defaultPath) => {
+export const normalizeSourcePath = (url = defaultPath) => {
+
+  // Process filepath resolution
+  //
   if (url.startsWith(".") || url.startsWith("/") || url.startsWith("~")) {
+    debug({ in: url, out: path.normalize(url)});
     return path.resolve(path.normalize(url));
   }
-  // remove the .git from the repo specifier
-  if (url.includes(".git")) {
-    url = url.replace(/.git$/, "");
-    url = url.replace(/.git#/, "#");
-    url = url.replace(/.git:/, ":");
-  }
 
-  // rewrite https://github.com/truffle-box/metacoin format in
-  //         https://github.com:truffle-box/metacoin format
-  if (url.match(/.com\//)) {
-    url = url.replace(/.com\//, ".com:");
-  }
+  // process https? or git prefixed input
+  //
+  if (/^(https?|git)/i.test(url) ){
+    const protocolRex = new RegExp(
+      "(?<protocol>(https://|git@))"  + 
+      "(?<service>[^@/]+)"            + // service
+      "(/|:)"                         + // separator. will be ignored in ouput
+      "(?<org>[^/#]+)"                + // org
+      "/"                             +
+      "(?<repo>[^/#]+)"               + // repo
+      "#?"                            + // optional branch separator
+      "(?<branch>[^#]+)?"             + // capture branch (undefined if unmatched)
```

Comment:
That's cool, I didn't know you could name these matches

Reply:
named capture groups are really useful!